src/redisearch_rs/rqe_iterators/src/metric.rs

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! Metric iterator implementation
+
+use crate::{RQEIterator, RQEIteratorError, SkipToOutcome, id_list::IdList};
+use ffi::{RSYieldableMetric, array_ensure_append_n_func, t_docId};
+use inverted_index::{RSIndexResult, ResultMetrics_Reset};
+use value::{RSValueFFI, RSValueTrait};
+
+pub enum MetricType {
+    VectorDistance,
+}
+
+/// An iterator that yields all ids within a given range, from 1 to max id (inclusive) in an index.
+pub struct Metric<'index> {
+    base: IdList<'index>,
+    metric_data: Vec<f64>,
+    #[allow(dead_code)]
+    type_: MetricType,
+}
+
+#[inline(always)]
+fn set_result_metrics(result: &mut RSIndexResult, val: f64) {
+    if let Some(num) = result.as_numeric_mut() {
+        *num = val;
+    } else {
+        // Safety: we created a metric result, which is numeric, in the constructor
+        panic!("Result is not numeric");
+    }
+
+    // SAFETY: reset the metrics c_array
+    unsafe {
+        ResultMetrics_Reset(result.metrics);
+    }
+
+    let value = RSValueFFI::create_num(val);
+    let new_metrics: *const RSYieldableMetric = &RSYieldableMetric {
+        key: std::ptr::null_mut(),
+        value: value.as_ptr(),
+    };
+    // SAFETY: calling a C function to append a new metric to the result's metrics array
+    unsafe {
+        result.metrics = array_ensure_append_n_func(
+            result.metrics as *mut std::ffi::c_void,
+            new_metrics as *mut std::ffi::c_void,
```

Comment:
```suggestion
            result.metrics as *mut _,
            new_metrics as *mut _,
```

You can use this less verbose form for such casts.